libgadget/blackhole.c

```diff
@@ -263,7 +273,15 @@ blackhole(ForceTree * tree)
     SphP_scratch->Injected_BH_Energy = mymalloc2("Injected_BH_Energy", SlotsManager->info[0].size * sizeof(MyFloat));
```

Comment:
Does it have to be mymalloc2?

Reply:
It does, unfortunately. Because we need to free SwallowID around it.

Reply:
nevermind. you didn't change this line.